src/main/java/org/synyx/urlaubsverwaltung/application/comment/ApplicationCommentServiceImpl.java

```diff
@@ -18,30 +23,52 @@
 class ApplicationCommentServiceImpl implements ApplicationCommentService {
 
     private final ApplicationCommentRepository commentRepository;
+    private final ApplicationService applicationService;
     private final Clock clock;
 
     @Autowired
-    ApplicationCommentServiceImpl(ApplicationCommentRepository commentRepository, Clock clock) {
+    ApplicationCommentServiceImpl(ApplicationCommentRepository commentRepository, ApplicationService applicationService,
+                                  Clock clock) {
+
         this.commentRepository = commentRepository;
+        this.applicationService = applicationService;
         this.clock = clock;
     }
 
     @Override
     public ApplicationComment create(Application application, ApplicationCommentAction action, Optional<String> text,
                                      Person author) {
 
-        final ApplicationComment comment = new ApplicationComment(author, clock);
-        comment.setAction(action);
-        comment.setApplication(application);
-        text.ifPresent(comment::setText);
+        final ApplicationCommentEntity commentEntity = new ApplicationCommentEntity(author, clock);
+        commentEntity.setAction(action);
+        commentEntity.setApplicationId(application.getId());
+        text.ifPresent(commentEntity::setText);
+
+        final ApplicationCommentEntity savedEntity = commentRepository.save(commentEntity);
+
+        final LongFunction<Application> byId = applicationId -> applicationService.getApplicationById(applicationId)
+            .orElseThrow(() -> new IllegalStateException("could not find application with id=" + applicationId));
 
-        return commentRepository.save(comment);
+        return toApplicationComment(savedEntity, byId);
     }
 
 
     @Override
     public List<ApplicationComment> getCommentsByApplication(Application application) {
-        return commentRepository.findByApplication(application);
+
+        final List<ApplicationCommentEntity> commentEntities = commentRepository.findByApplicationId(application.getId());
+
+        final List<Long> applicationIds = commentEntities
+            .stream()
+            .map(ApplicationCommentEntity::getApplicationId)
+            .distinct()
+            .toList();
+
+        final Map<Long, Application> applicationById = applicationService.findApplicationsByIds(applicationIds)
+            .stream()
+            .collect(Collectors.toMap(Application::getId, Function.identity()));
+
+        return commentEntities.stream().map(entity -> toApplicationComment(entity, applicationById::get)).toList();
```

Comment:
kann es Kommentare ohne Urlaubsantrag geben? Soll man hier noch defensiver sein falls in der `Map` kein Wert gefunden wird? 🤔

Reply:
ich würde sagen nein, mir fällt gerade keins ein.